packages/terra-time-input/src/TimeInput.jsx

```diff
@@ -724,30 +720,195 @@ class TimeInput extends React.Component {
+     * @returns {String} a description of the hour field, intended to be read to screen reader users.
+     */
+    function hourDescription() {
+      if (is12Hour) {
+        return intl.formatMessage({
+          id: 'Terra.timeInput.hourDescriptionTwelve',
+          defaultMessage: 'A two-digit 12-hour value',
+          description: `Explains to screen reader users that the hour field needs a two digit hour. This will be read
+          only when a screen reader is enabled. It is never displayed. It will be read when the user has focused on
+          the hour input or when the screen reader is reading the page to the user.`,
+        });
+      }
+      return intl.formatMessage({
+        id: 'Terra.timeInput.hourDescriptionTwentyFour',
+        defaultMessage: 'A two-digit 24-hour value',
+        description: 'Like Terra.timeInput.hourDescriptionTwelve but for a 24-hour clock.',
+      });
+    }
+
+    // Fan out some component-level props into input-level a11y attributes. See the big comment below for more info.
+    inputAttributes.isInvalid = isInvalid;
+    inputAttributes.disabled = disabled;
+    inputAttributes.required = required;
+
+    const a11yString = TimeUtil.getA11YTimeValue(this.props, this.state, this.postMeridiem);
+
+    function groupLabel() {
+      if (label) {
+        return intl.formatMessage({ // fix
+          id: 'Terra.timeInput.inputGroupValue',
+          defaultMessage: '{label}',
+          description: `This will be read by screen readers as the reader moves into the group of inputs. It is intended to
+      help the user understand "you are about to enter a section of the page where many different inputs all work
+      together for this one concept of time.`,
+        }, { label });
+      }
+      return intl.formatMessage({
+        id: 'Terra.timeInput.inputGroupValueDefault',
+        defaultMessage: 'Time',
+        description: `Same meaning as groupLabel, only for situations where the consumer has not provided a label for us
+      to plug in.`,
+      });
+    }
+
+    function hoursLabel() {
+      if(label){
+        return intl.formatMessage({
+          id: 'Terra.timeInput.hourLabel',
+          defaultMessage: '{label} hour',
+          description: `The label that will only be read to screen readers. It is prefixed with the time input's name,
+          e.g. 'Time of Birth', so that screen reader users can pick this specific hour field out of a list of many
+          hour fields on the same page. The minute and second screen reader labels won't contain the name because they
+          will always follow their labeld hour field. We didn't want to say the label too many times.`
+        }, {label});
+      }
+      return intl.formatMessage({
+        id: 'Terra.timeInput.hourLabelDefault',
+        defaultMessage: 'Hour',
+        description: `Like Terra.timeInput.hourLabel but used in situations where the consumer of the time input did not
```

Comment:
I feel like referencing the id of another message in the description could get confusing if the messages are ever not right next to each other.

Reply:
CTRL+F is always there? What would you suggest as an alternative.